### PR TITLE
Added FK to flowcharts table + Linting

### PR DESCRIPTION
**Pull request by jayshrivastava**
- I added a FK relation from the flowcharts table to the flowchart_node table (adding a column for the root node which is nullable so we can tell if a flowchart is empty). 
- Schema auto updated from running the migration
- I ran rubocop to auto lint my updates

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -47,4 +49,5 @@
   add_foreign_key "flowchart_nodes", "flowchart_nodes", column: "child_id_id"
   add_foreign_key "flowchart_nodes", "flowchart_nodes", column: "sibling_id_id"
   add_foreign_key "flowchart_nodes", "flowcharts", column: "flowchart_id_id"
+  add_foreign_key "flowcharts", "flowchart_nodes", column: "root_id_id"
```
[the remark this answers is not in the document]
Comment (jayshrivastava) [reply db/schema.rb]: I think we should. I updated all of the migrations.

File: db/schema.rb [hunk 2]
```diff
@@ -37,14 +37,17 @@
...
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
   end
 
-  add_foreign_key "flowchart_nodes", "flowchart_nodes", column: "child_id_id"
-  add_foreign_key "flowchart_nodes", "flowchart_nodes", column: "sibling_id_id"
-  add_foreign_key "flowchart_nodes", "flowcharts", column: "flowchart_id_id"
+  add_foreign_key "flowchart_nodes", "flowchart_nodes", column: "child_id"
+  add_foreign_key "flowchart_nodes", "flowchart_nodes", column: "sibling_id"
+  add_foreign_key "flowchart_nodes", "flowcharts"
```
Comment (lee-ma) [inline db/schema.rb]: What does this line do? Just wondering

Comment (jayshrivastava) [reply db/schema.rb]: My best guess is the column name is implicitly `tablename_id` so it does not write the column in the schema file. For the other columns we added, we haven't named them `tablename_id`, so it explicitly puts them.

I checked the schema for the `flowchart_nodes` table and it looks right. 
image
image

Comment (lee-ma) [reply db/schema.rb]: cool, looks good then

File: db/migrate/20191105020642_create_flowchart_node.rb [hunk 3]
```diff
@@ -7,10 +7,10 @@ def change
       t.string :header, null: false
       t.string :button_text
       t.string :next_question
-      t.references :child_id, foreign_key: { to_table: :flowchart_nodes }
```
Comment (ddgwilli) [inline db/migrate/20191105020642_create_flowchart_node.rb]: For next time, you can short hand these calls through the `belongs_to` associations.

**Pull request merged**